nmostesting/TestHelper.py

```diff
@@ -266,10 +267,99 @@ def get_messages(self):
         return msg_cpy
 
     def did_error_occur(self):
-        return self.error_occured
+        return self.error_occurred
 
     def get_error_message(self):
         return self.error_message
 
     def clear_messages(self):
         self.messages.clear()
+
+
+class MQTTClientWorker:
+    """MQTT Client Worker"""
+    def __init__(self, host, port, secure=False, username=None, password=None, topics=[]):
+        """
+        Initializer
+        :param host: broker hostname (string)
+        :param port: broker port (int)
+        :param secure: use TLS (bool)
+        :param username: broker username (string)
+        :param password: broker password (string)
+        :param topics: list of topics to subscribe to (list of string)
+        """
+        self.host = host
+        self.port = port
+        self.error_occurred = False
+        self.connected = False
+        self.error_message = ""
+        # MQTT 5 is required so that we can set retainAsPublished
+        # when subscribing to test whether messages have retain flags set
+        self.client = mqtt.Client(protocol=mqtt.MQTTv5)
+        self.client.on_connect = lambda client, userdata, flags, rc, properties=None: self.on_connect(flags, rc)
+        self.client.on_disconnect = lambda client, userdata, rc: self.on_disconnect(rc)
+        self.client.on_message = lambda client, userdata, msg: self.on_message(msg)
+        self.client.on_subscribe = lambda client, userdata, mid, *args: self.on_subscribe(mid)
+        self.client.on_log = lambda client, userdata, level, buf: self.on_log(level, buf)
+        if secure:
+            self.client.tls_set(CONFIG.CERT_TRUST_ROOT_CA)
+        if username or password:
+            self.client.username_pw_set(username, password)
+        self.topics = topics
+        self.pending_subs = set()
+        self.messages = []
+
+    def start(self):
+        self.client.connect_async(self.host, self.port)
+        self.client.loop_start()
+
+    def close(self):
+        self.client.loop_stop()
+
+    def on_connect(self, flags, rc):
+        if len(self.topics) == 0:
+            self.connected = True
+        else:
+            for topic in self.topics:
+                result, message_id = self.client.subscribe(topic, options=mqtt.SubscribeOptions(retainAsPublished=True))

```

Comment:
In the current implementation
https://github.com/AMWA-TV/nmos-testing/pull/557/files#diff-1e75e5acaa0fcbec3701ba4d454cdc3aR503
will add topics which are not checked for `None`:
```
                for sender in broker_senders[broker_params]:
                    topics.add(sender.connection_status_topic)
                    topics.add(sender.topic)
```
Certainly `connection_status_topic` is allowed to be `None`, so then the MQTT client will attempt to subscribe to a topic of `None` which will trigger an MQTT exception `MQTT log: 8: Caught exception in on_connect: No topic specified, or incorrect topic type.`

I would potentially check the topics for `None` when adding to the set and then trigger appropriate responses (e.g. the sender topic should never be `None`)

Reply:
Thanks Christian, should be fixed now

Reply:
Thanks! Works great now.